lib/puppet/provider/file/windows.rb

```diff
@@ -86,6 +88,32 @@ def validate
     end
   end
 
+  # munge the windows group permissions if the group is set to SYSTEM
+  #
+  # tl;dr: when SYSTEM user is the group and the resoure is not managing group then treat
+  # the resource as insync if System has FullControl access.
+  #
+  # On windows systems: files laid down when puppet is running as the SYSTEM user will have
+  # the SYSTEM user set as the "group" on them. In these situations we want puppet to ignore
+  # what the permissions should be as long as the system user has FullControl access.
+  def munge_windows_system_group(current, should)
+    group_fullcontrol = "070".to_i(8)
+    if resource[:group].nil? && (group == Puppet::Util::Windows::SID::LocalSystem)
+      if (current.to_i(8) & group_fullcontrol) == group_fullcontrol
+        # Since the group is LocalSystem, and the permissions are FullControl,
+        # replace the value returned by group ownership with the value expected
+        # by 'should'. This will treat this specific situation as "insync"
+        current = ((current.to_i(8) & "707".to_i(8)) | (should[0].to_i(8) & "070".to_i(8))).to_s(8).rjust(4, '0')
+      else
+        # If the SYSTEM account does _not_ have FullControl in this scenario, we should
+        # force the resource out of sync no matter what.
+        Puppet.debug _("Not managing SYSTEM user and SYSTEM user does not have full control, treating mode as out of sync")
```

Comment:
The message here is a bit misleading.
`Debug: Not managing SYSTEM user and SYSTEM user does not have full control, treating mode as out of sync`
Which resource?
Is this the owner or group?
I feel the msg needs more context to make it actionable for a user.

Also may need to add a note for Translators to NOT translate the word SYSTEM as that's a literal word